website/blog/2021-07-28-release-apache-apisix-2.8-en.md

```diff
@@ -0,0 +1,143 @@
+
+The following features are currently included:
+
+- Dynamically set mTLS
+- Dynamically set client_max_body_size
+- Upstream keepalive （2.8 new feature）
+- gzip （2.8 new Plug-in）
+
+In future releases, we will continue to allow the following Nginx configurations to be set dynamically:
+
+- real_ip
+- proxy_max_temp_file_size
+- ……
+
+An example of Upstream configuration:
+
+```JSON
+{
+    "id": "backend",
+    "nodes": {"host:80": 100},
+    "type":"roundrobin",
+    "keepalive_pool": {
+        "size": 4,
+        "idle_timeout": 8,
+        "requests": 16
+    }
+}
+```
+
+### New experience: Enhance stream proxy
+
+In version 2.8, the [ip-restriction](http://apisix.apache.org/docs/apisix/plugins/ip-restriction/) and [limit-conn](http://apisix.apache.org/docs /apisix/plugins/limit-conn/) had been duplicated from the HTTP section to the stream section. The benefit of this way is to enhance gateway capabilities in the stream proxy and to increase the security of the upstream services.
+
+ip-restriction can be used to filter IP black and white list to ensure that only requests from a specific IP can access the backend service.
+
+limit-conn can be used to limit the number of simultaneous connections on a route, limiting the number of concurrent client accesses.
+
+### New plug-in: gzip
+
+Apache APISIX version 2.8 contains the gzip plug-in. Using the gzip plug-in, you can **dynamically set route-level gzip parameters**.
+
+An example of gzip configuration:
+
+```JSON
+{
+    "plugins": {
+        "gzip": {
+            "min_length": 20,
+            "http_version": 1.1,
+            "buffers": {
+                "number": 32,
+                "size": 4096
+            },
+            "types": [
+                "text/html"
+            ],
+            "comp_level": 1,
+            "vary": false
+        }
+    }
+}
+```
+
+### New plug-in: ua-restriction
+
+The `ua-restriction` plugin is used to check if the User-Agent is in the black and white list, which is a very common requirement and can be enabled by way of a plugin.
+
+An example of `ua-restriction` configuration:
+
+```JSON
+{
+    "plugins": {
+        "ua-restriction": {
+            "denylist": [
+                "my-bot1",
+                "(Baiduspider)/(\\d+)\\.(\\d+)"
+            ]
+        }
+    }
+}
+```
+
+### New way to develop: Support for executing specific logic by plug-ins
+
+Based on Apache APISIX architecture, many features are implemented by plug-ins. Starting from version 2.8, **Apache APISIX supports executing specific logic by plug-ins after selecting an upstream node.**
+
+Define the following method in the plug-in:
+
+```Lua
+function _M.balancer(conf, ctx)
+    core.log.notice("IP: ", ctx.balancer_ip, ", Port: ", ctx.balancer_port)
+end
+```
+
+In this example, the log will output the IP and Port of the upstream.
+
+**Which scenario will use the above method?**
```

Comment:
scenario does not use the method.
Suggestion: --> Which scenario does the above method apply to?